File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 9/16/2025, 12:11:02 AM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        
        min_length = min(len(s) for s in strs)
        pre = ''
        
        for i in range(min_length):
            char_at_pos = {s[i] for s in strs}
            if len(char_at_pos) == 1:
                pre += strs[0][i]  # Add the common character to the prefix
            else:
                break  # Stop if characters differ
        
        return pre

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.cpp
// Last updated: 9/16/2025, 12:10:57 AM
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";

        int i = 0;

        while (true) {
            if (i >= strs[0].size()) break;  // Reached end of first string
            char c = strs[0][i];

            for (int j = 1; j < strs.size(); j++) {
                if (i >= strs[j].size() || strs[j][i] != c) {
                    return strs[0].substr(0, i); 
                }
            }

            i++;  
        }

        return strs[0].substr(0, i);
    }
};
